superduperdb/ext/openai/model.py

```diff
@@ -95,29 +95,29 @@ async def _apredict_one(self, X: str, **kwargs):
         return e['data'][0]['embedding']
 
     @retry
-    def _predict_a_batch(self, texts, **kwargs):
-        out = Embedding.create(input=texts, model=self.identifier, **kwargs)['data']
-        return [r['embedding'] for r in out]
+    def _predict_a_batch(self, texts: t.List[str], **kwargs):
+        out = Embedding.create(input=texts, model=self.identifier, **kwargs)
+        return [r['embedding'] for r in out['data']]
 
     @retry
-    async def _apredict_a_batch(self, texts, **kwargs):
-        out = await Embedding.acreate(input=texts, model=self.identifier, **kwargs)[
-            'data'
-        ]
-        return [r['embedding'] for r in out]
+    async def _apredict_a_batch(self, texts: t.List[str], **kwargs):
+        out = await Embedding.acreate(input=texts, model=self.identifier, **kwargs)
+        return [r['embedding'] for r in out['data']]
 
-    def _predict(self, X, batch_size=100, **predict_kwargs):
+    def _predict(self, X, one: bool = False, **kwargs):
```

Comment:
Preciously existing line 108 right after this is just awful practice. "On `str`, throw these arguments away and do this other thing!"

But I'm not understanding the change to the signature here.  Moving `batch_size` out of the signature to  `batch_size = kwargs.pop('batch_size', 100)` makes it less clear.  This parameter `one` is never used!!

Reply:
I agree that there are issues here.

The motivation is that the 'interface' is defined [here](https://github.com/nenb/superduperdb/blob/f247eef896195eac48a600763a91d33547bab00b/superduperdb/container/model.py#L369). Every implementation that uses `PredictMixin` (like this one) will be called by this interface.

Ideally, I would modify the interface in that link. But it has impacts elsewhere, and I didn't want to tackle it in this PR. So, I decided that I would at least make sure that we are following a standard interface, even if it looks a bit weird.

As an example, before this change, `one` was getting lumped inside `**predict_kwargs** and getting sent to the OpenAI API, even though `one` isn't a valid query parameter for this API (I think they just swallow invalid query parameters...)

Does this make sense? Or am I on a wild goose chase here?

Reply:
Oh, dear, I even knew about this. GAH.

Absolutely, there is no great way to deal with this and this is more accurate than what was there before...